### ISSUE TITLE
Column summaries - toggle between linear and log base 10

### ISSUE THREAD

**Issue by julie-sullivan** [intermine/im-tables#176]
From Gos:

It would be great to have a toggle button when viewing the little graphs, that
would switch from the Y-axis being linear and it being log base 10.

Good example would be: 

ThaleMine -> 
  view lists -> 
    List Analysis for Workshop Sucrose Transporters List (9 Genes) 
    Scroll to bottom and open Genes --> Expression (33,384) rows
       --> column summary of Expression Signal.

A log plot would show the distribution better...

**Comment by yochannah** [comment]
See https://github.com/intermine/intermine/issues/1891 for background

**Comment by CodHeK** [comment]
@yochannah Can I try this ?

**Comment by yochannah** [comment]
@CodHeK feel free to take a look! 

as it stands we are getting the data from InterMine and just displaying it, so we'd theoretically need to _either_ wait for the intermine endpoint to be updated _or_ we'd need to figure our how to transform the data on our own. If the data transform is complex it's probably better to leave to the server endpoint rather than putting it on the client side. Let me know what you find!

**Comment by CodHeK** [comment]
> Good example would be:
> 
> ThaleMine ->
> view lists ->
> List Analysis for Workshop Sucrose Transporters List (9 Genes)
> Scroll to bottom and open Genes --> Expression (33,384) rows
> --> column summary of Expression Signal.

@julie-sullivan could you please explain how to view this example a little more ... [this](https://apps.araport.org/thalemine/begin.do) is the site we're querying at right ?

**Comment by CodHeK** [comment]
@julie-sullivan following the links you added ... I couldn't find `Gene --> Expression` in that page ...
This is how it looks ...

![Screenshot 2019-04-01 at 3 58 40 PM](https://user-images.githubusercontent.com/22097366/55321435-453e0c80-5497-11e9-8edc-b02ab75d59be.png)

**Comment by yochannah** [comment]
@CodHeK I think this issue here is that the original wishlist log was made in 2015, so things have changed a little. Try clicking on RNA-seq Expression and look at the column summary for the 

RNA Seq Expressions
Expression Level

column to get an idea of the problem - a few columns are visible but most are really small and can't be compared easily.

**Comment by CodHeK** [comment]
@yochannah did a bit of research here ... I believe [this](https://github.com/intermine/im-tables/blob/master/src/views/facets/histogram.coffee#L13) is where things need to be changed ... d3 has a `d3.scale.log()`  function that can do the work and condition need to be applied depending on the distribution fetched from the intermine when queried ...

**Comment by yochannah** [comment]
that sounds great - defaulting to whichever looks better and also allowing users to switch from one to the other would probably make Gos (our boss) really happy ;)

**Comment by CodHeK** [comment]
@yochannah if you could hint me where is the web service call happening such as the one you mentioned 

```
GET someintermine.org/service/query=<query model="genomic" view="Gene.pathways.name Gene.primaryIdentifier Gene.length" sortOrder="Gene.pathways.name ASC" ></query>&summaryPath=Gene.length&format=jsonrows
```

**Comment by yochannah** [comment]
@CodHeK where it happens in imtables / imjs?

**Comment by CodHeK** [comment]
@yochannah yup, I mean the place where the query is fetched from the warehouse

**Comment by yochannah** [comment]
oh, I see what you mean. This should be part of the implementation of the imtables library and not pary of the query at all. Setting log vs linear maybe should be done using a ui control.

**Comment by yochannah** [comment]
le me know if that makes sense! :grin:

**Comment by CodHeK** [comment]
Ah, right! So, this has nothing to do with querying right! So, the data is there, it has been fetched ... the chart is made I just have add an option to toggle the X-axis right ?

**Comment by yochannah** [comment]
Precisely!

**Comment by CodHeK** [comment]
@julie-sullivan @yochannah the scales present here are :

```
scales = # return:
      x: (scale buckets, xPositions) # A scale from bucket -> x
      y: (scale counts, yPositions)  # A scale from count -> y
      valToX: (scale values, xPositions) # A scale from value -> x
      xToVal: (toVal xPositions) # A scale from x -> value
      bucketToVal: (toVal buckets) # A scale from bucket -> min val
```
here we need the `y-axis` values to be `log` based right  ? I am little confused here ...
